File: Lab2.REST/Models/Song.cs
using System.ComponentModel.DataAnnotations;

namespace MusicWebAPI.Models
{
    public class Song
    {
        [Key]
        public int Id { get; private set; }

        [Required]
        public int Year { get; set; }

        [Required]
        public string Title { get; set; }

        [Required]
        [Range(0, int.MaxValue)]
        public int Duration { get; set; }

        public string? Lyrics { get; set; }

        public int AuthorId { get; set; }

        public Artist? Author { get; set; }

    }
}

File: Lab2.GraphQL/src/models/Song.ts
import { Entity, Column, BaseEntity } from "typeorm";
import { ObjectType, Field, ID } from "type-graphql";
import { Artist } from "./Artist";
// import { v4 as uuidv4 } from 'uuid'; 

@ObjectType()
@Entity()
export class Song extends BaseEntity {
  @Field(() => ID)
  id: string;

  @Field()
  @Column()
  title: string;

  @Field()
  @Column()
  year: number;

  @Field()
  @Column()
  duration: number;

  @Field(() => Artist)
  author: Artist;

  constructor(title: string, id: string, year: number, duration: number, author: Artist) {
    super();
    // this.id = uuidv4(); 
    this.id = id
    this.title = title;
    this.duration = duration;
    this.year = year;
    this.author = author;
  }
}
